src/turnkeyml/common/build.py

```diff
@@ -297,10 +328,52 @@ def get_wmic_info(command):
                 info_dict["Physical Memory"] = f"{mem_info_gb} GB"
             except ValueError:
                 info_dict["Physical Memory"] = mem_info_bytes
+            info_dict["BIOS Version"] = get_wmic_info("wmic bios get smbiosbiosversion")
```

Comment:
wmic was deprecated recently and in most runs we are not getting this information unless the user installs wmic. 
But Get-WmiObject is built into PowerShell (up to version 5.1). In PowerShell 7 and beyond, it is replaced by Get-CimInstance.
Can you please replace all the wmic and Get-WmiObject commands with the following Get-CimInstance alternatives:

**Processor**: Get-CimInstance -ClassName Win32_Processor | Select-Object Name, Manufacturer, NumberOfCores, NumberOfLogicalProcessors
**OEM info**: Get-CimInstance -ClassName Win32_ComputerSystem | Select-Object Manufacturer, Model
**Physical memory**: Get-CimInstance -ClassName Win32_PhysicalMemory | Select-Object Manufacturer, Capacity, Speed
**BIOS info**: Get-CimInstance -ClassName Win32_BIOS | Select-Object SMBIOSBIOSVersion
**CPU Max clock**: Get-CimInstance -ClassName Win32_Processor | Select-Object MaxClockSpeed

Reply:
One thought is to add a global argument flag to skip gathering the system info.  For example, the command `lemonade -i Qwen/Qwen2.5-0.5B-Instruct oga-load --device igpu --dtype int4` takes about 22 seconds to run, assuming that the model is already cached locally.  If you skip gathering the system info, the same command takes less than 6 seconds to run.  I can imagine many situations where you only need to gather the system info once (it is in fact sticky in the turnkeyml_stats.yaml file), so would be happy to save the time in most calls to lemonade.  How about adding a `-x` global flag to skip gathering the system info?

Reply:
Turns out there is an API that does that really well and is not discontinued

import wmi


class SystemInfo:
    def __init__(self):
        # Initialize WMI connection
        self.connection = wmi.WMI()

    def get_processor_name(self):
        """
        Retrieves the name of the processor.

        Returns:
            str: Name of the processor.
        """
        processors = self.connection.Win32_Processor()
        if processors:
            return processors[0].Name
        return "Processor information not found."

    def get_system_model(self):
        """
        Retrieves the model of the computer system.

        Returns:
            str: Model of the computer system.
        """
        systems = self.connection.Win32_ComputerSystem()
        if systems:
            return systems[0].Model
        return "System model information not found."


# Example Usage
if __name__ == "__main__":
    system_info = SystemInfo()
    print("Processor Name:", system_info.get_processor_name())
    print("System Model:", system_info.get_system_model())

Reply:
Using the wmi api as Daniel has shown above might be the fastest way.

Reply:
Thanks Daniel - that's great!  It looks like this will be 20% quicker than the original implementation calling `wmic`.  I will update the code.  Since it will be easiest to implement using a class, I will refactor the code and put it in its own `system_info.py` file.